server/service/osquery.go

```diff
@@ -973,6 +1001,13 @@ func (svc *Service) SubmitDistributedQueryResults(
 		if err := svc.task.RecordPolicyQueryExecutions(ctx, host, policyResults, svc.clock.Now(), ac.ServerSettings.DeferredSaveHost); err != nil {
 			logging.WithErr(ctx, err)
 		}
+	} else {
+		if hostWithoutPolicies {
```

Comment:
Nit, but it would remove a level of nesting by combining the `else` and `if` into `else if hostWithoutPolicies`.